Makefile

```diff
@@ -12,6 +12,12 @@ include $(config)
 export LDFLAGS = -pthread -lm
 export CFLAGS = -std=c++11 -Wall -O2 -Iinclude -fPIC
 
+ifdef DMLC_CORE_PATH
```

Comment:
Maybe I am misunderstanding something:  in config.mk, it has the string ADD_CFLAGS= with no string after it.  It is harmless for a blank ADD_CFLAGS to be expanded in Makefile, but this isn't the case with DMLC_CORE_PATH, since the include selection follows from whether DMLC_CORE_PATH override is defined.  This would mean that config.mk would need to read:

# the additional link flags you want to add
ADD_LDFLAGS=

# the additional compile flags you want to add
ADD_CFLAGS=

# path to dmlc-core module 
DMLC_CORE_PATH=NONE


and not:

# the additional link flags you want to add
ADD_LDFLAGS=

# the additional compile flags you want to add
ADD_CFLAGS=

# path to dmlc-core module 
DMLC_CORE_PATH=